src/main/java/startsteps/ECommerceShop/service/JwtServiceImpl.java

```diff
@@ -64,7 +63,7 @@ public boolean isTokenValid(String token, UserDetails userDetails) {
     }
 
     private boolean isTokenNotExpired(String token) {
-        return extractExpiration(token).before(new Date());  // have to check logic???
+        return extractExpiration(token).before(new Date());
```

Comment:
If the expiration date is before the current date, it means the token has expired. Are you sure this should be your logic?